src/Analysis/Problem/Fabric/FabricModDependencyProblem.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Aternos\Codex\Minecraft\Analysis\Problem\Fabric;
+
+use Aternos\Codex\Minecraft\Analysis\Solution\Forge\ModInstallSolution;
+use Aternos\Codex\Minecraft\Translator\Translator;
+
+/**
+ * Class ModDependencyProblem
+ *
+ * @package Aternos\Codex\Minecraft\Analysis\Problem\Fabric
+ */
+class FabricModDependencyProblem extends FabricModProblem
+{
+    /**
+     * @var string
+     */
+    protected static $modNamePattern = "('([^\']+)' \(([^\)]+)\)|(\w+))";
+
+    /**
+     * @var string
+     */
+    protected static $modIDPattern = "([^ ,]+)";
+
+    /**
+     * @var array
+     */
+    protected $dependency;
+
+    /**
+     * Get a human readable message
+     *
+     * @return string
+     */
+    public function getMessage(): string
+    {
+        $mods = "'" . $this->getDependency() . "'";
+
+        return Translator::getInstance()->getTranslation("mod-dependency-problem", ["mod-name" => $this->getModName(), "dependency-mod-name" => $mods]);
+    }
+
+    /**
+     * Get an array of possible patterns
+     *
+     * The array key of the pattern will be passed to setMatches()
+     *
+     * @return array
+     */
+    public static function getPatterns(): array
+    {
+        return [
```

Comment:
You can use keys (e.g. short descriptive strings) in this array, which you can then use in `setMatches()` to identify which pattern was used. This is easier to read and also makes it easier to change/add patterns or extend the class.